Mothercell/Hive.IO/Hive.IO/Plots/OperationalPerformancePlot.cs

```diff
@@ -0,0 +1,103 @@
+        private bool _normalized = false;  // the state we start out in
+        private readonly EnergyPlotProperties _properties;
+        private RectangleF _bounds = RectangleF.Empty;
+
+        /// <summary>
+        /// Draw a square box with data inside and a unit string above.
+        /// </summary>
+        /// <param name="totalHeight">the total height available including unit string</param>
+        /// <param name="properties">EnergyPlotProperties to use for this plot</param>
+        public OperationalPerformancePlot(EnergyPlotProperties properties)
+        {
+            _standardFont = GH_FontServer.Standard;
+            _boldFont = GH_FontServer.StandardBold;
+            _properties = properties;
+            _penWidth = 30f;
+        }
+
+        private String UnitText => _normalized ? _properties.NormalizedUnitText : _properties.UnitText;
+
+        private String Data(Results results)
+        {
+            var value = _normalized ? _properties.NormalizedData(results) : _properties.Data(results);
+            return $"{value:F1}";
+        }
+
+        public bool Contains(PointF location) => _bounds.Contains(location);
+
+        public void Clicked(GH_Canvas sender)
+        {
+            _normalized = !_normalized;
+            sender.Invalidate();
+        }
+
+        /// <summary>
+        /// Draw a box with some text
+        /// The bold text (Data) goes bang in the middle, centered vertically and horizontally.
+        /// The standard text (Unit) goes in the lower part, bang in the middle, centered horizontally
+        /// </summary>
+        /// <param name="results"></param>
+        /// <param name="graphics"></param>
+        /// <param name="bounds">expected height of bounds should be TotalHeight</param>
+        public void Render(Results results, Graphics graphics, RectangleF bounds)
+        {
+            // save bounds for Contains method - when user clicks on the plot
+            _bounds = bounds;
+            var brush = new SolidBrush(Color.Black);
+            var pen = new Pen(_properties.Color, _penWidth);
+
+            // draw box
+            var box = new RectangleF(bounds.Location, bounds.Size);
+            box.Inflate(-_penWidth / 2, -_penWidth / 2); // make sure lines fit _inside_ box
+            graphics.DrawRectangle(pen, box.Left, box.Top, box.Width, box.Height);
+            if (!_normalized)
+            {
+                graphics.FillRectangle(new SolidBrush(_properties.Color), box);
+            }
+
+            // center data in the box
+            var data = Data(results);
+            var dataSize = GH_FontServer.MeasureString(data, _boldFont);
+            var dataX = bounds.Left + (bounds.Width - dataSize.Width) / 2;
+            var dataY = bounds.Top + bounds.Height / 2 - (float)dataSize.Height / 2;
+            graphics.DrawString(data, _boldFont, brush, dataX, dataY);
+            RhinoApp.WriteLine($"Rendering {data} to {dataX}, {dataY}");
```

Comment:
`RhinoApp.WriteLine($"Rendering {data} to {dataX}, {dataY}");`

I would deactivate this. It spams the command line in Rhino every time you move the Visualizer component or change its size etc. Usually, you wanna keep the rhino command line clean so you can track the actual Rhino commands that you've used for drawing geometry

Reply:
good point. I'll delete all the debug output - was just using that for creating the tool...

Reply:
ok cool. will you merge this branch then when you're done? cheers